llvm/lib/CodeGen/GlobalISel/LegalizerHelper.cpp

```diff
@@ -3690,7 +3690,7 @@ LegalizerHelper::LegalizeResult LegalizerHelper::lowerLoad(GAnyLoad &LoadMI) {
 
   if (!isPowerOf2_32(MemSizeInBits)) {
     // This load needs splitting into power of 2 sized loads.
-    LargeSplitSize = llvm::bit_floor(MemSizeInBits);
+    LargeSplitSize = llvm::bit_floor(MemSizeInBits.getKnownMinValue());
```

Comment:
These values are used in the call to getMachineMemOperand. There is a signature:
```
MachineMemOperand *MachineFunction::getMachineMemOperand(
    const MachineMemOperand *MMO, const MachinePointerInfo &PtrInfo, LLT Ty)
```
It might be safer to create a scalable MMO using `Ty` rather than forcing it into fixed.

Ditto for the other instructions that are built with `LargeSplitSize` and `SmallSplitSize`.

Reply:
Do you mean we should use `LLT LargeSplitSize, SmallSplitSize` instead of `uint64_t`?

Reply:
I think so. Otherwise we risk loosing out on the fact that the size is actually scalable.